libs/common/src/helpers/facility-calculations/index.test.ts

```diff
@@ -0,0 +1,26 @@
+import { calculateDrawnAmount, calculateInitialUtilisation } from '.';
+
+describe('helpers/facility-calculations', () => {
+  const mockFacilityValue = 150000;
+  const mockCoverPercentage = 20;
+
+  describe('calculateInitialUtilisation', () => {
+    it('should return correct calculation', () => {
+      const result = calculateInitialUtilisation(mockFacilityValue);
+
+      const expected = mockFacilityValue * 0.1;
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('calculateDrawnAmount', () => {
+    it('should return correct calculation', () => {
+      const result = calculateDrawnAmount(mockFacilityValue, mockCoverPercentage);
+
+      const expected = calculateInitialUtilisation(mockFacilityValue) * (mockCoverPercentage / 100);
+
+      expect(result).toEqual(expected);
```

Comment:
I do not agree with this.  We have the confidence from the `calculateInitialUtilisation` tests and this helper uses the returned value from that function.

Reply:
The `* (mockCoverPercentage / 100)` is duplicating implementation rather than actually testing the correct value is returned. It would be better to mock the 'calculateInitialUtilisation` helper in this test so that we can make it return a set known value in this test and then we can make the test better as suggested.

Reply:
I agree with @Zainzzkk 

1. Hard coding is not a good practice.
2. `calculateInitialUtilisation` is tested.
3. `calculateDrawnAmount` is simply calling `calculateInitialUtilisation` with some certain values (2 of which are "dynamic" / from params).
4. We only need to test that `calculateInitialUtilisation` is calling `calculateDrawnAmount` with the expected params. That is exactly what the function/implementation does, I do not see why it should be tested any differently.
5. If the implementation changes, this test will correctly fail.
6. If we mock the helper and the implementation changes - depending on the mock, the test might still pass.
7. I don't see any reason to mock helpers in any unit test. I have never seen such a pattern in any project; With the expection for API/DB calls etc.

The current test is perfect IMO.

Reply:
I disagree a few counter points:

1.  "We only need to test that  `calculateInitialUtilisation` is calling `calculateDrawnAmount `with the expected params." - This test isn't actually testing this, it is testing that the outcome of the function under test is _equal_ to if you had called this function then done some modifications to the result. If you want to test this function is _actually_ you could spy on it and check it is actually called.

2. `(mockCoverPercentage / 100)` is used in the test and in the function itself. If this part fo the calculation didn't do what was intended for some reason then the test would still pass. The point of a test is to fail if the function doesn't work as intended by the developer. 
3. "I don't see any reason to mock helpers in any unit test. I have never seen such a pattern in any project; With the expection of API/DB calls etc." - I've got the exact opposite experience, all the projects I have worked on have always in cases like this mocked the helper method instead of calling it to set up the expected values.

Reply:
If that still doesn't convince you I will resolve the thread in the interest of this getting merged but I really disagree with this and would like us to come to a place of agreement for future work.